datafusion/core/tests/physical_optimizer/push_down_filter.rs

```diff
@@ -278,7 +273,7 @@ fn test_filter_collapse() {
         -     DataSourceExec: file_groups={0 groups: []}, projection=[a, b, c], file_type=test, pushdown_supported=true
       output:
         Ok:
-          - DataSourceExec: file_groups={0 groups: []}, projection=[a, b, c], file_type=test, pushdown_supported=true, predicate=b@1 = bar AND a@0 = foo
+          - DataSourceExec: file_groups={0 groups: []}, projection=[a, b, c], file_type=test, pushdown_supported=true, predicate=a@0 = foo AND b@1 = bar
```

Comment:
This actually looks like an improvement to me as now `a = foo` will be evaluated before `b=bar` as was done in the input plan. This might be important for short circuiting, perhaps

The prior version of this optimization seems to have reordered them